### DISCUSSION TITLE
Why do engines NEED to be in the "Engines" directory in FlightGear?

### DISCUSSION THREAD

**Discussion by Octal450** [JSBSim-Team/jsbsim#1169] [Support requests]
Hi,
It seems I MUST place engines in the "Engines" directory in FlightGear JSBSim FDE. Anywhere else, and JSBSim "can't find" the file even if I specify the path like I do for every other thing.

Why is this?

Kind Regards,
Josh

**Comment by seanmcleod** [comment]
So here is the logic in JSBSim for looking for an engine file.

https://github.com/JSBSim-Team/jsbsim/blob/57361d0756a7a4d477d80f786e381317d8722671/src/models/FGPropulsion.cpp#L440-L461

Basically it first looks for the engine file, e.g. `CFM56` in a `Engine` or `Engines` sub-directory of the aircraft, see lines 455-458.

If that fails, then via line 460 it looks for it in the specified `EnginePath`.

Now `EnginePath` in JSBSim stand-alone:

https://github.com/JSBSim-Team/jsbsim/blob/57361d0756a7a4d477d80f786e381317d8722671/src/JSBSim.cpp#L359-L363

Where the `root` path can be specified via the command line, otherwise it defaults to the current directory.

`--root=<path>  specifies the JSBSim root directory (where aircraft/, engine/, etc. reside)`

Now, when it comes to FlightGear they may setup `EnginePath` differently. Although even if they do, you should still be able to bundle your engine file in a `engine` sub-directory of the aircraft.

For example here a snippet from `FlightGear.cxx` that lives in the JSBSim repo, although note that this was last updated 7 years ago, so you would need to double-check FlightGear's current source code.

https://github.com/JSBSim-Team/jsbsim/blob/57361d0756a7a4d477d80f786e381317d8722671/examples/FlightGear.cxx#L224-L230

**Comment by Octal450** [comment]
Thanks for the explanation.

I guess I'll look at getting this fixed in FG.

J
